### PR TITLE
Updated workflow to accompany the multiple-curation design of `element-calcium-imaging` - using `session` element - Added test suite

### PR DESCRIPTION
**Pull request by ttngu207**

### CODE REVIEW

File: tests/__init__.py [hunk 1]
```diff
@@ -0,0 +1,643 @@
...
+        'imaging_root_data_dir': os.environ.get('IMAGING_ROOT_DATA_DIR', dj.config['custom']['imaging_root_data_dir'])
+    }
+    return
+
+
+@pytest.fixture
+def pipeline():
+    from workflow_imaging import pipeline
+
+    yield (pipeline.subject, pipeline.lab, pipeline.imaging,
```
Comment (ixcat) [inline tests/__init__.py:29]: might be easier/more straightforward to yield a dictionary & use the **kwargs mechanism in client code
e.g: `yield {'subject': pipeline.subject, ...}`, or potentially define this as a dictionary within pipeline itself and then e.g. `yield pipeline.modules`

File: tests/test_ingest.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import pathlib
+
+from . import *
```
Comment (ixcat) [inline tests/test_ingest.py:3]: we typically avoid `import *` - 
* some linter tools cant pin down imported symbols as used/unused
* reader doesn't know where some symbols come from (esp. if multiple * used)
* risk of overriding/colliding silently if upstream dependencies shift.

**Review by ixcat — APPROVED** [review]
was excited to see the pipeline testing approach, so took a look/left a few comments.
take as you like; nothing seems critical in my view

File: tests/__init__.py (hunk 1 above)
Comment (ixcat) [reply tests/__init__.py:29]: ... although, thinking further - this also requires other functions to be aware of the dictionary contents & name accordingly (e.g. no other 'subject' argument, etc) ...

File: tests/test_ingest.py (hunk 2 above)
Comment (ttngu207) [reply tests/test_ingest.py]: agreed, not a big fan of `import *` either. I had it there for quick prototyping then forgot to change. PR updated now for the fix

Comment (ixcat) [reply tests/test_ingest.py]: sounds good - thought you tended not to, so apologies if preachy/condescending sounding; just saving potential for back & forth :+1:

File: workflow_imaging/ingest.py [hunk 3]
```diff
@@ -30,18 +30,15 @@ def ingest_sessions():
```
Comment (kabilar) [inline workflow_imaging/ingest.py:30]: Can remove the above declaration of `processing_tasks`.

**Review by dimitri-yatsenko — CHANGES_REQUESTED** [review]

**Comment by kabilar** [comment]
Pull request will be merged to move forward.  Subsequent pull request will be open to address any issues.

**Pull request merged**